### PR TITLE
Retry tests with Spek

### PR DESCRIPTION
**Pull request by skyylex**
**Description**

One more PR with unit tests. Now it's Retry tested by Spek. 

**Notes**

Spek works quit nice, haven't found anything that blocks us from using it. If everyone agrees to use Spek, I'd convert other unit tests.

### CODE REVIEW

File: Tests/PumaTests/RetryTests.swift [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                            XCTFail("Failure result isn't expected")
+                        case .success(_):
+                            expectation.fulfill()
+                        }
+                    }
+                    
+                    self.wait(for: [expectation], timeout: 0.1)
+                }
+                
+                It("should complete with a failure with 2 attempts and task that fails 3 time") {
```
Comment (skyylex) [inline Tests/PumaTests/RetryTests.swift]:
```suggestion
                It("should complete with a failure after 2 attempts for task that fails 3 times") {
```

File: Tests/PumaTests/RetryTests.swift [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                            XCTFail("Failure result isn't expected")
+                        case .success(_):
+                            expectation.fulfill()
+                        }
+                    }
+                    
+                    self.wait(for: [expectation], timeout: 0.1)
+                }
+                
+                It("should complete successfully with 3 attempts and task that fails 2 time") {
```
Comment (skyylex) [inline Tests/PumaTests/RetryTests.swift]:
```suggestion
                It("should complete successfully after 3 attempts for task that fails 2 times") {
```

File: Tests/PumaTests/RetryTests.swift [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                            expectation.fulfill()
+                        case .success(_):
+                            XCTFail("Success result isn't expected")
+                        }
+                    }
+                    
+                    self.wait(for: [expectation], timeout: 0.1)
+                }
+                
+                It("should complete successfully with 0 attempts and task that succeeds") {
```
Comment (skyylex) [inline Tests/PumaTests/RetryTests.swift]:
```suggestion
                It("should complete successfully when no attempts allowed and task that succeeds") {
```

File: Tests/PumaTests/RetryTests.swift [hunk 4]
```diff
@@ -0,0 +1,100 @@
+import Foundation
+import Spek
+import XCTest
+@testable import PumaCore
+
+final class RetryTests: XCTestCase {
+    func testSpec() {
+        spec {
+            Describe("Retry") {
+                It("shouldn't retry with 0 attempts and task that fails") {
```
Comment (skyylex) [inline Tests/PumaTests/RetryTests.swift]:
```suggestion
                It("should complete with a failure when no attempts allowed and task that fails once") {
```

File: Tests/PumaTests/RetryTests.swift [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+                    let retry = Retry(times: 0) { MockedTask(numberOfFailures: 1) }
+                    let expectation = self.expectation(description: "Expect failure result in completion")
+                    retry.run(workflow: Workflow()) { result in
```
Comment (onmyway133) [inline Tests/PumaTests/RetryTests.swift]: I should have add a sugar in Spek to make these expectation code easier

File: Tests/PumaTests/RetryTests.swift [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+                }
+            
+            }
+        }
+    }
+}
+
+private extension RetryTests {
+    class MockedTask: Task {
+        var name: String = "A victim task of Retry"
```
Comment (onmyway133) [inline Tests/PumaTests/RetryTests.swift]: 😎

**Comment by onmyway133** [comment]
@skyylex nice work, you can make another PR with your provided suggestions if you want

**Pull request merged**